theano/gpuarray/multinomial.py

```diff
@@ -327,6 +327,7 @@ def gpu_kernels(self, node, name):
 
                     if (! %(replace)s )
                     {
+                        __syncthreads();
```

Comment:
I let decide which one you prefer. I don't have the head to evaludate the code completly today.